remme/atomic_swap_tp/client.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2018 REMME
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import json
+import datetime
+
+import logging
+
+from remme.atomic_swap_tp.handler import AtomicSwapHandler
+from remme.protos.atomic_swap_pb2 import AtomicSwapInitPayload, AtomicSwapExpirePayload, AtomicSwapClosePayload, \
+    AtomicSwapMethod, AtomicSwapInfo, AtomicSwapSetSecretLockPayload, AtomicSwapApprovePayload
+from remme.protos.certificate_pb2 import NewCertificatePayload, CertificateMethod
+from remme.settings import GENESIS_ADDRESS, SETTINGS_SWAP_COMMISSION, ZERO_ADDRESS
+from remme.settings.helper import _make_settings_key
+from remme.shared.basic_client import BasicClient
+from remme.certificate_tp.handler import CertificateHandler
+from remme.shared.utils import attr_dict
+from remme.token_tp.handler import TokenHandler
+
+LOGGER = logging.getLogger(__name__)
+
+
+@attr_dict
```

Comment:
Is this really required? For me it looks like totally not required layer of additional logic.

Reply:
it is a syntactical sugar to avoid dictionary brackets for each line

Reply:
Why whouldn't you just use `**kwargs` or named arguments?

Reply:
Oh, well, I see the way you use them, and there is one simple drawback: you can just declare named arguments like `func(a=something, b=something_else)` and then use it like that:
```python
args = { 'a': 'value1', 'b': 'value2' }
func(**args)
```

Such a method also gives you less room for a mistake. With the current implementation of `AttrDict` you can simply forget to specify one of the parameters and it will five you `AttributeError`, while with named arguments you can always use default values.

I am strictly against duplicating the existing functionality in any fashion.